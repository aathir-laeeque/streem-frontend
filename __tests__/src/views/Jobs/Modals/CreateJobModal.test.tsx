import {
  CreateJobModal,
  CreateJobModalProps,
} from '#views/Jobs/Modals/CreateJobModal';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { create, ReactTestRenderer } from 'react-test-renderer';

let snapshot: ReactTestRenderer;

const props: CreateJobModalProps = {
  closeAllModals: jest.fn(),
  closeModal: jest.fn(),
  selectedChecklist: {
    code: '123',
    id: 1,
    name: 'Test Checklist',
    version: 1,
    properties: {
      test_property: 'Test Property',
    },
  },
  properties: [
    {
      id: 1,
      name: 'test_property',
      placeHolder: 'Test Placeholder',
      orderTree: 1,
      mandatory: true,
    },
    {
      id: 2,
      name: 'test_property2',
      placeHolder: 'Test Placeholder 2',
      orderTree: 2,
      mandatory: false,
    },
  ],
  onCreateJob: jest.fn(),
};

beforeAll(() => {
  const modal = <CreateJobModal {...props} />;
  snapshot = create(modal);
});

describe('<CreateJobModal />', () => {
  it('it matches the snapshot', () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it('should have a checklistId', () => {
    render(<CreateJobModal {...props} />);
    const inputNode = screen.getByTestId('checklistId');
    expect(inputNode).toBeTruthy();
    expect(inputNode).toHaveProperty(
      'value',
      `${props.selectedChecklist.code} ${props.selectedChecklist.name}`,
    );
  });

  it.each(props.properties)('Properties Rendered Successfully', (property) => {
    render(<CreateJobModal {...props} />);
    expect(screen.getByTestId(property.name)).toBeTruthy();
  });

  it('Should Change to Error State for Leaving Manadory Property Input Empty', () => {
    render(<CreateJobModal {...props} />);
    const inputNode = screen.getByTestId(props.properties[0].name);
    inputNode.focus();
    inputNode.blur();
    expect(inputNode).toHaveStyle('border-color: #ff6b6b');
  });

  it('Should Not Change to Error State for Leaving Non-Manadory Property Input Empty', () => {
    render(<CreateJobModal {...props} />);
    const inputNode = screen.getByTestId(props.properties[1].name);
    inputNode.focus();
    inputNode.blur();
    expect(inputNode).toHaveStyle('border-color: #666666');
  });
});
