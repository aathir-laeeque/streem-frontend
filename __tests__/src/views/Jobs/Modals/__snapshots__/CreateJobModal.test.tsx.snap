// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateJobModal /> it matches the snapshot 1`] = `
<div
  className="sc-fzoyAV lgNqAT"
>
  <div
    className="sc-fznyAO ctDhaX"
  >
    <div
      className="openup"
      id="modal-container"
    >
      <div
        className="modal-background"
      >
        <div
          className="modal"
        >
          <div
            className="modal-header"
          >
            <h2>
              Creating a Job
            </h2>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "fontSize": 20,
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          <div
            className="modal-body"
          >
            <div
              className="sc-fzokOt daYHNr"
            >
              <div
                className="wrapper"
              >
                <label
                  className="on"
                  style={
                    Object {
                      "color": "#666666",
                      "opacity": 0.4,
                    }
                  }
                >
                  Checklist
                </label>
                <input
                  className="input"
                  data-testid="checklistId"
                  disabled={true}
                  name="checklistId"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="Checklist"
                  style={
                    Object {
                      "borderColor": "#666666",
                      "opacity": 0.4,
                    }
                  }
                  type="text"
                  value="123 Test Checklist"
                />
              </div>
            </div>
            <div
              className="sc-fzokOt daYHNr"
            >
              <div
                className="wrapper"
              >
                <label
                  className=""
                  style={
                    Object {
                      "color": "#666666",
                    }
                  }
                >
                  test_property
                </label>
                <input
                  className="input"
                  data-testid="test_property"
                  disabled={false}
                  name="test_property"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="test_property"
                  style={
                    Object {
                      "borderColor": "#666666",
                      "opacity": 1,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="sc-fzokOt daYHNr"
            >
              <div
                className="wrapper"
              >
                <label
                  className=""
                  style={
                    Object {
                      "color": "#666666",
                    }
                  }
                >
                  test_property2
                </label>
                <input
                  className="input"
                  data-testid="test_property2"
                  disabled={false}
                  name="test_property2"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  placeholder="test_property2"
                  style={
                    Object {
                      "borderColor": "#666666",
                      "opacity": 1,
                    }
                  }
                  type="text"
                  value=""
                />
                <span
                  className="optional-text"
                >
                  Optional
                </span>
              </div>
            </div>
          </div>
          <div
            className="modal-footer"
          >
            <div
              className="modal-footer-options"
            >
              <span
                style={
                  Object {
                    "color": "#12aab3",
                    "fontSize": 12,
                    "fontWeight": 600,
                  }
                }
              >
                Schedule Job
              </span>
            </div>
            <div
              className="modal-footer-buttons"
            >
              <button
                className="sc-fznZeY fIRQkD"
                onClick={[Function]}
                style={
                  Object {
                    "fontWeight": 600,
                    "padding": "5px 16px",
                  }
                }
              >
                Cancel
              </button>
              <button
                className="sc-fznKkj dICuB"
                onClick={[Function]}
                style={
                  Object {
                    "fontWeight": 600,
                    "marginRight": 0,
                  }
                }
              >
                Create Job
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
